framework/generated/generated_vulkan_replay_consumer.cpp

```diff
@@ -9203,18 +9202,15 @@ void VulkanReplayConsumer::Process_vkCmdDrawMeshTasksIndirectCountEXT(
     GetDeviceTable(in_commandBuffer)->CmdDrawMeshTasksIndirectCountEXT(in_commandBuffer, in_buffer, offset, in_countBuffer, countBufferOffset, maxDrawCount, stride);
 }
 
-template <typename T>
-void InitializeOutputStructPNext(StructPointerDecoder<T> *decoder)
+void InitializeOutputStructPNext(size_t len, const void* input, void* output, size_t struct_type_size)
 {
-    if(decoder->IsNull()) return;
-    size_t len = decoder->GetOutputLength();
-    auto input = decoder->GetPointer();
-    auto output = decoder->GetOutputPointer();
-    for( size_t i = 0 ; i < len; ++i )
+    if (!input || !output) return;
+
+    for (size_t i = 0 ; i < len; ++i)
     {
-        const auto* in_pnext = reinterpret_cast<const VkBaseInStructure*>(input[i].pNext);
-        if( in_pnext == nullptr ) continue;
-        auto* output_struct = reinterpret_cast<VkBaseOutStructure*>(&output[i]);
+        const auto* in_pnext = reinterpret_cast<const VkBaseInStructure*>(static_cast<const uint8_t*>(input) + i * struct_type_size)->pNext;
+        if (in_pnext == nullptr) continue;
```

Comment:
If this a problem, it was a problem in the original code too, but are we leaving an uninitialized hole in the output array if we hit a null ptr on input here? Should we zero the pnext of the output at least?